spec/unit/plugins/linux/kernel_spec.rb

```diff
@@ -31,15 +31,40 @@
 virtio_balloon         13168  0
 floppy                 55441  0
 ENV_LSMOD
+    @version_module = {
+      dm_crypt: "",
+      psmouse: "",
+      acpiphp: "",
+      microcode: "1.2.3",
+      serio_raw: "",
+      virtio_balloon: "",
+      floppy: "",
+    }
+
+    @expected_result = {
+      "dm_crypt"       => { "size" => "22321", "refcount" => "0" },
+      "psmouse"        => { "size" => "81038", "refcount" => "0" },
+      "acpiphp"        => { "size" => "23314", "refcount" => "0" },
+      "microcode"      => { "size" => "18286", "refcount" => "0", "version" => "1.2.3" },
+      "serio_raw"      => { "size" => "13031", "refcount" => "0" },
+      "virtio_balloon" => { "size" => "13168", "refcount" => "0" },
+      "floppy"         => { "size" => "55441", "refcount" => "0" },
+    }
     @plugin = get_plugin("kernel")
     allow(@plugin).to receive(:collect_os).and_return(:linux)
     allow(@plugin).to receive(:init_kernel).and_return({})
     allow(@plugin).to receive(:shell_out).with("uname -o").and_return(mock_shell_out(0, "Linux", ""))
     allow(@plugin).to receive(:shell_out).with("env lsmod").and_return(mock_shell_out(0, @env_lsmod, ""))
+    @version_module.each do |mod, vers|
+      allow(@plugin).to receive(:run_command).with({ command: "cat /sys/module/#{mod}/version", :no_status_check => true }).and_return([1, vers, ""])
+    end
     expect(@plugin).to receive(:shell_out).with("env lsmod").at_least(1).times
     @plugin.run
   end
 
   it_should_check_from_deep_mash("linux::kernel", "kernel", "os", "uname -o", [0, "Linux", ""])
 
+  it "Testing linux::kernel::modules" do
```

Comment:
The string here should be a description of the test. For example, `it "collects linux::kernel::modules" do`.